sequencer/src/lib.rs

```diff
@@ -374,6 +389,14 @@ pub async fn init_node<P: PersistenceOptions, Ver: StaticVersionType + 'static>(
         }
     };
 
+    let version = Ver::version();
+    if let Some(upgrade) = genesis.upgrades.get(&version) {
+        let view = upgrade.view;
+        config.config.start_proposing_view = view;
+        config.config.stop_proposing_view = view + 1;
```

Comment:
I was testing this but I am going to ask Salam about this

Reply:
I was testing this but I am going to ask Salman about this

Reply:
In the local demo it's probably fine because views rarely or never fail. But in production with nodes stopping and starting and real world network conditions, it's very possible we could fail a view at the wrong time

Reply:
should we make it u64::MAX? We keep progressing and proposing the upgrade until it is done? @ss-es

Reply:
or maybe we provide a window like 1000 views and if the upgrade isn't proposed then something is really wrong?

Reply:
Have we tried it recently with a range? I think setting `stop_proposing_view = view + 1000` should be completely fine (I think it's unlikely you'll need more than a 10 view range actually)

> This means we have one view to propose it and if we aren't successful it just fails?

Yes, but I think it's slightly harder to fail an upgrade than it is to just fail a view. The proposal is broadcast separately from the QuorumProposal, and goes out several views in advance from when we attach it. After we get enough votes to form a certificate, we will attach it to our own proposal and it should be viral at that point: any node that sees it is supposed to take it and start re-attaching it until the upgrade completes, or fails. (This is the intention at least, there might be some bugs in the code.)

Even with a 1-view window, to fail the upgrade we need to fail to collect enough votes on it by the time we propose (currently we start voting 5 views ahead iirc, but we can make this configurable), or we need a failure where nobody sees our proposal.

I'm currently working through some serialization issues that are blocking getting any of this merged into main, but I can double check that the multiple-view window works as intended on the HotShot side soon.

Reply:
Sounds good. In that case I think setting a small reasonable window (10 views) sounds good. We probably don't want it to propose indefinitely, if something goes wrong it's probably actually better to stop, so we can investigate and try again

Reply:
added propose_window to the config ef2e4913018cf3d088423e8b364d0e4f3ff87485